in_toto/verifylib.py

```diff
@@ -306,294 +308,582 @@ def verify_all_steps_command_alignment(layout, links_dict):
     verify_command_alignment(command, expected_command)
 
 
-def verify_match_rule(rule, artifact_queue, artifacts, links):
-  """
-  <Purpose>
-    Verifies that target path pattern - 3rd or 5th element of rule - matches
-    at least one file in target artifacts. This might conflict with
-    general understanding of glob patterns (especially "*").
+def _equal_hash_dicts(hash_dict_a, hash_dict_b):
+  """Compare to hash_dict for equality, formats are
+  securesystemslib.formats.HASHDICT_SCHEMA. Return True or False. """
```

Comment:
Is this check being made before? or should we run this check before doing anything else.